File: Code/Components/DanpheEMR.ServerModel/LabModels/LabRequisitionModel.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace DanpheEMR.ServerModel
{

    public class LabRequisitionModel
    {

        [Key]
        public Int64 RequisitionId { get; set; }
        public int? PatientVisitId { get; set; }
        public int PatientId { get; set; }
        public int? ProviderId { get; set; }
        public Int64 LabTestId { get; set; }
        public string ProcedureCode { get; set; }
        public string LOINC { get; set; }
        public string LabTestName { get; set; }
        public string LabTestSpecimen { get; set; }
        public string LabTestSpecimenSource { get; set; }
        public string PatientName { get; set; }
        public string Diagnosis { get; set; }
        public string Urgency { get; set; }
        public DateTime? OrderDateTime { get; set; }
        public string ProviderName { get; set; }
        public string BillingStatus { get; set; }
        public string OrderStatus { get; set; }
        public int? SampleCode { get; set; }
        public string RequisitionRemarks { get; set; }
        public DateTime? SampleCreatedOn { get; set; }
        public int? SampleCreatedBy { get; set; }
        public string Comments { get; set; }
        public string RunNumberType { get; set; }   

        public List<LabTestComponentResult> LabTestComponentResults { get; set; }

        //required for billing for listing the imaging items <dinesh:19Jan'17>
        public virtual LabTestModel LabTest { get; set; }
        public virtual PatientModel Patient { get; set; }

        //Suraj: Updated to not null
        public int ReportTemplateId { get; set; }

        public int? DiagnosisId { get; set; }

        public DateTime? CreatedOn { get; set; }
        public int? CreatedBy { get; set; }
        public DateTime? ModifiedOn { get; set; }
        public int? ModifiedBy { get; set; }

        public bool? IsActive { get; set; }//sud:15Sept'18--to exclude InActive Records.

        //added: ashim : 18Sep2018
        public string VisitType { get; set; }
        public int? LabReportId { get; set; }
        public int? BarCodeNumber { get; set; }
        public string WardName { get; set; }

        public bool? IsVerified { get; set; }
        public DateTime? VerifiedOn { get; set; }
        public int? VerifiedBy { get; set; }

        public int ResultingVendorId { get; set; }//sud:16May'19--for lab external vendors.
        public bool? HasInsurance { get; set; }
    }

    public class LabTestTransactionItemVM
    {
        public Int64 BillItemPriceId { get; set; }
        public int ServiceDepartmentId { get; set; }
        public string ServiceDepartmentName { get; set; }
        public string ServiceDepartmentShortName { get; set; }
        public int? Displayseq { get; set; }
        public int ItemId { get; set; }
        public string ItemName { get; set; }
        public string ProcedureCode { get; set; }
        public int Price { get; set; }
        public bool TaxApplicable { get; set; }
        public bool DiscountApplicable { get; set; }
        public string Description { get; set; }
        public string RunNumberType { get; set; }
    }

}

File: Code/Websites/DanpheEMR/wwwroot/DanpheApp/src/app/labs/shared/lab-requisition.model.ts
import * as moment from "moment";
import { ENUM_DateTimeFormat } from "../../shared/shared-enums";

export class LabTestRequisition {
  public RequisitionId: number = 0;
  public PatientId: number = 0;
  public PatientVisitId: number = 0;
  //Ashim: 4Jan2018
  //ProviderId is RequestedBy Dr. from billing/ doctor who gives lab order in Doctors
  // public ProviderId: number = 0;
  public PrescriberId: number = null; // Krishna, 15th,jun'22 , changed ProviderId to PrescriberId.
  public LabTestId: number = 0;
  public ProcedureCode: string = null;
  public LOINC: string = null;
  public LabTestName: string = null;
  public LabTestSpecimen: string = null;
  public LabTestSpecimenSource: string = null;
  public PatientName: string = null;
  public Diagnosis: string = null;
  public Urgency: string = "Normal";//setting it as a default
  public OrderDateTime: string = null;
  // public ProviderName: string = null;
  public PrescriberName: string = null; // Krishna, 15th,jun'22 , changed ProviderName to PrescriberName.
  public BillingStatus: string = null;
  public OrderStatus: string = null;
  public SampleCode: string = null;
  public RequisitionRemarks: string = null;
  public CreatedBy: number = 0;
  public CreatedOn: string = moment().format(ENUM_DateTimeFormat.Year_Month_Day_Hour_Minute);
  public SampleCreatedBy: number = null;
  public SampleCreatedOn: string = null;
  public SampleCollectedOnDateTime: string = null;
  public Comments: string = "";
  public ReportTemplateId: number = 0;
  public RunNumberType: string = null;
  public BillCancelledBy: number = null;
  public BillCancelledOn: string = null;
  public LabTypeName: string = null;

  //Added by Anish While making Diagnosis
  public DiagnosisId: number = 0;
  //added: ashim: 18Sep2018
  public VisitType: string = null;
  public LabReportId: number = null;
  public WardName: string = null;
  public IsActive: boolean = true;

  public IsVerified: boolean = false;
  public VerifiedBy: number = null;
  public VerifiedOn: string = null;

  public ResultingVendorId: number = 0;
  public HasInsurance: boolean = false;
  public IsSmsSend: boolean = false;

  public IsSelected: boolean = false;
  public ServiceItemId: number = 0;
}

